data/marshaller.go

```diff
@@ -41,20 +42,16 @@ func Marshal(w http.ResponseWriter, r *http.Request, subject interface{}) {
 		w.Header().Set(httpHeaderContentType, contentTypeXML)
 		w.Write(resp)
 		return
-	} else {
-		convertTo = "json"
-		resp, err = json.Marshal(subject)
-		w.Header().Set(httpHeaderContentType, contentTypeJSON)
-		w.Write(resp)
-		return
 	}
-
+	convertTo = "json"
+	resp, err = json.Marshal(subject)
```

Comment:
Please move that back and remove the return in line 43/44. It prevents errors from being handled.